### PR TITLE
Refactor and clean up ScreenShot handling code and improve device support

### LINKED ISSUE #325: Fix instructions for screenshot mode
Splitting from #273 (which came from #217), since it didn't get attention there:

> Change instructions when selecting battery saver mode (instructions continue to say "Press IV button" when it should be "take a screenshot", might confuse people who forgot they have turned the option on and then don't see the button anywhere).

@DJCrapsody Do you think you can fix this by 3.1.0?

### PR DESCRIPTION
**Pull request by sarav**
Moved all ScreenShot handling code into a separate class. This also fixes battery saver mode not working for a lot of devices (surprisingly, there's not issue for this. Or at least I couldn't find one).

Also fixes #325 indirectly.

### CODE REVIEW

**Comment by sarav** [comment]
@Blaisorblade wanna review this?

The left over commits us

**Comment by sarav** [comment]
*The left over commit is just deleting code from MainActivity and adding the throttling of the event handling.

**Comment by sarav** [comment]
@Blaisorblade @farkam135 

One of you want to review this?

**Comment by sarav** [comment]
Tested the last commit too. Works well for me.

File: app/src/main/java/com/kamron/pogoiv/ScreenShotHelper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class ScreenShotHelper {
+
+    private static ScreenShotHelper instance = null;
+    private FileObserver screenShotScanner;
+
+    private ScreenShotHelper(final Context context, final String screenShotDir) {
+        screenShotScanner = new FileObserver(screenShotDir, FileObserver.CLOSE_NOWRITE | FileObserver.CLOSE_WRITE) {
+            @Override
+            public void onEvent(int event, String file) {
+                if (file != null) {
```
Comment (Blaisorblade) [inline app/src/main/java/com/kamron/pogoiv/ScreenShotHelper.java]: Memo: This commit stops using `readyForNewScreenshot` silently. Hopefully later commits clean this up.

File: app/src/main/java/com/kamron/pogoiv/ScreenShotHelper.java [hunk 2]
```diff
@@ -1,47 +1,81 @@
...
+                if (!pathChange.contains("Screenshot")) {
+                    return;
+                }
+
+                try {
+                    Bitmap bitmap = BitmapFactory.decodeFile(pathChange);
+                    Intent newintent = Pokefly.createProcessBitmapIntent(bitmap, pathChange);
                     LocalBroadcastManager.getInstance(context).sendBroadcast(newintent);
+                } catch (Exception e) {
+                    // TODO: Retry a few times after a wait
```
Comment (Blaisorblade) [inline app/src/main/java/com/kamron/pogoiv/ScreenShotHelper.java]: Memo: this should also be addressed somewhere.

**Comment by Blaisorblade** [comment]
Review finished. Works for me too on Nexus 5—though the delay is quite noticeable. Merging.

**Comment by sarav** [comment]
Phew! Finally merged. Is this noticeably slower than the file observer method? Any idea if the Nexus generates one or multiple content observer events? 

Another option is to have both observers going, but then file observer is apparently a memory hog. I guess this is what you get for battery saver mode I guess.

**Comment by Blaisorblade** [comment]
> Is this noticeably slower than the file observer method?

Just now it feels noticeably slower than non-battery-saver, haven't really compared with the file observer.

> Any idea if the Nexus generates one or multiple content observer events?

Nope. I guess I'll need to set a breakpoint to catch this. We'll see.

**Comment by sarav** [comment]
Yeah, battery save mode has always been slower than normal mode. I think the content observer notifications get send after adding the image to the Media data base, etc. So, might be even more slower. I'm sure even with FileObserver, it'll take more time since it needs to be written to disk and then read back (or hopefully cached) instead of directly copying from display buffer.

File: app/src/main/java/com/kamron/pogoiv/ScreenShotHelper.java (hunk 1 above)
Comment (sarav) [inline app/src/main/java/com/kamron/pogoiv/ScreenShotHelper.java]: I'm assuming you saw the later commit that just deletes that flag? We don't really need that flag in addition to the ones in Pokefly. This is part of the effort to reduce it to 1 flag. Something like "isBusyWithUser" that denotes that the IV related UI is active. When that happens, we shouldn't need to process any screens or screenshots.

But then, it's going to be really hard to trigger that anyway. Even if you take a screenshot, the pokemon scan code won't see any pokemon and bail.

**Pull request merged**